Makefile

```diff
@@ -103,10 +106,11 @@ USBFS_DIR       = $(ROOT)/lib/main/STM32_USB-FS-Device_Driver
 USBPERIPH_SRC   = $(notdir $(wildcard $(USBFS_DIR)/src/*.c))
 FATFS_DIR       = $(ROOT)/lib/main/FatFS
 FATFS_SRC       = $(notdir $(wildcard $(FATFS_DIR)/*.c))
+SEGGER_RTT_DIR  := $(ROOT)/lib/main/segger
```

Comment:
Looks like this also needs the library to be added to the repository.

Reply:
Doh!

Reply:
There are two main uses of this; one is to hook up the CLI to run over the debug connection, not USB, which will enable CLI commands to be issued whilst the GUI is doing other things, and the other is logging. I've implemented a few logging frameworks over the years, so I know what's needed here. I'll set to work on it over the next few days. The same logging mechanism will also be applicable to use over a serial link if desired.